source/github/webhook.go

```diff
@@ -77,8 +77,12 @@ func (c *client) VerifyWebhook(request *http.Request, r *library.Repo) error {
 // processPushEvent is a helper function to process the push event.
 func processPushEvent(h *library.Hook, payload *github.PushEvent) (*types.Webhook, error) {
 	logrus.Tracef("processing push GitHub webhook for %s", payload.GetRepo().GetFullName())
-
+	logrus.Tracef("THIS IS THE COMMENT BEFORE THE GOOD STUFF")
+	logrus.Tracef("Payload: %s", payload)
 	repo := payload.GetRepo()
+	pusherName := payload.GetPusher().Name
+	pusherEmail := payload.GetPusher().Email
```

Comment:
You likely want to utilize the [sender](https://github.com/google/go-github/blob/master/github/event_types.go#L824) since the [pusher](https://github.com/google/go-github/blob/master/github/event_types.go#L823) can incorrectly point to a different user if someone sends a commit with a different email/name with valid credentials. You can test the difference between pusher and sender by sending commits with `git commit --author="John Doe <john@doe.org>"`.

Reply:
Interesting... the sender does not have an email field. Do you think it is fine to leave that out?

Reply:
Do you have an issue that describes what this is trying to solve? I've confirmed that tag events within Vela 0.9 have this data available within the environment variables of `VELA_BUILD_SENDER`, `VELA_BUILD_AUTHOR`, and `VELA_BUILD_AUTHOR_EMAIL`.

Reply:
[https://github.com/go-vela/community/issues/436](url) Sorry it was an internal issue that wasn't duplicated yet. So since the `VELA_BUILD_SENDER` already exists, the only missing information is the email, which as I mentioned does not come with the payload. Any ideas? Should we just inform the user to use the `VELA_BUILD_SENDER` for the name and scrap the email? The `BUILD_AUTHOR_EMAIL` will not work since it is tied to the last person to commit. Not sure if it's a good idea to set the `BUILD_AUTHOR` and `BUILD_AUTHOR_EMAIL` to the tagger on tag events, but that is an option. Let me know!